subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -94,9 +105,20 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 				       buf, sizeof(buf))));
 		}
 
-		net_if_ipv6_addr_add(context->iface,
-				     (struct in6_addr *)(&address->mAddress),
-				     NET_ADDR_ANY, 0);
+		if_addr = net_if_ipv6_addr_add(
+					context->iface,
+					(struct in6_addr *)(&address->mAddress),
+					NET_ADDR_AUTOCONF, 0);
+
+		if (if_addr == NULL) {
+			NET_WARN("Not enough buffers to register OpenThread's"
```

Comment:
Why "buffers"?

Reply:
And this is not NET_WARN I guess.

Reply:
> Why "buffers"?

Because of lack of better word for "IP address placeholder"

> And this is not NET_WARN I guess.

And what exactly is wrong with NET_WARN?

Reply:
> Because of lack of better word for "IP address placeholder"

The right word would be "entry", but the whole message is problematic. That message is printed when net_if_ipv6_addr_add() returns NULL. Are you sure that the only condition when it does that is when there're no free addres entries? Are you sure it'll stay like that in the future? "Cannot add OpenThread <if possible, type> address" is the message which will never lie. (I'm all for more detailed error messages, as I said, but there're limits to it all.)

> And what exactly is wrong with NET_WARN?

Because it's *error*. Unless that address is not used for anything useful, and we add it just for fun ;-).

Reply:
Ok, I'll update this piece